pkg/tsdb/sqleng/sql_engine_test.go

```diff
@@ -9,317 +9,308 @@ import (
-				So(fixtures[4].(int64), ShouldEqual, tMilliseconds)
-				So(fixtures[5].(int64), ShouldEqual, tMilliseconds)
-				So(fixtures[6], ShouldBeNil)
-			})
-		})
+		fixtures := make([]interface{}, 3)
+		fixtures[0] = tSeconds
+		fixtures[1] = &tSeconds
+		fixtures[2] = nilInt
 
-		Convey("Given row values with uin64 as time columns", func() {
-			tSeconds := uint64(dt.Unix())
-			tMilliseconds := uint64(dt.UnixNano() / 1e6)
-			tNanoSeconds := uint64(dt.UnixNano())
-			var nilPointer *uint64
-
-			fixtures := make([]interface{}, 7)
-			fixtures[0] = tSeconds
-			fixtures[1] = &tSeconds
-			fixtures[2] = tMilliseconds
-			fixtures[3] = &tMilliseconds
-			fixtures[4] = tNanoSeconds
-			fixtures[5] = &tNanoSeconds
-			fixtures[6] = nilPointer
-
-			for i := range fixtures {
-				ConvertSqlTimeColumnToEpochMs(fixtures, i)
-			}
+		for i := range fixtures {
+			ConvertSqlTimeColumnToEpochMs(fixtures, i)
+		}
 
-			Convey("When converting them should return epoch time with millisecond precision ", func() {
-				So(fixtures[0].(int64), ShouldEqual, tSeconds*1e3)
-				So(fixtures[1].(int64), ShouldEqual, tSeconds*1e3)
-				So(fixtures[2].(int64), ShouldEqual, tMilliseconds)
-				So(fixtures[3].(int64), ShouldEqual, tMilliseconds)
-				So(fixtures[4].(int64), ShouldEqual, tMilliseconds)
-				So(fixtures[5].(int64), ShouldEqual, tMilliseconds)
-				So(fixtures[6], ShouldBeNil)
-			})
+		t.Run("When converting them should return epoch time with millisecond precision", func(t *testing.T) {
+			require.Equal(t, dt.Unix()*1e3, fixtures[0].(int64))
+			require.Equal(t, dt.Unix()*1e3, fixtures[1].(int64))
+			require.Nil(t, fixtures[2])
 		})
+	})
 
-		Convey("Given row values with int32 as time columns", func() {
-			tSeconds := int32(dt.Unix())
-			var nilInt *int32
-
-			fixtures := make([]interface{}, 3)
-			fixtures[0] = tSeconds
-			fixtures[1] = &tSeconds
-			fixtures[2] = nilInt
-
-			for i := range fixtures {
-				ConvertSqlTimeColumnToEpochMs(fixtures, i)
-			}
-
-			Convey("When converting them should return epoch time with millisecond precision ", func() {
-				So(fixtures[0].(int64), ShouldEqual, dt.Unix()*1e3)
-				So(fixtures[1].(int64), ShouldEqual, dt.Unix()*1e3)
-				So(fixtures[2], ShouldBeNil)
-			})
+	t.Run("Given row values with float64 as time columns", func(t *testing.T) {
+		tSeconds := float64(dt.UnixNano()) / float64(time.Second)
+		tMilliseconds := float64(dt.UnixNano()) / float64(time.Millisecond)
+		tNanoSeconds := float64(dt.UnixNano())
+		var nilPointer *float64
+
+		fixtures := make([]interface{}, 7)
+		fixtures[0] = tSeconds
+		fixtures[1] = &tSeconds
+		fixtures[2] = tMilliseconds
+		fixtures[3] = &tMilliseconds
+		fixtures[4] = tNanoSeconds
+		fixtures[5] = &tNanoSeconds
+		fixtures[6] = nilPointer
+
+		for i := range fixtures {
+			ConvertSqlTimeColumnToEpochMs(fixtures, i)
+		}
+
+		t.Run("When converting them should return epoch time with millisecond precision", func(t *testing.T) {
```

Comment:
Do you mind just removing t.Run here? Every t.Run introduces a sub-test, and I don't think it makes sense to group assertions as sub-tests. IMO a sub-test should correspond to a test case.